### PR TITLE
Add group roles to UserInfo interface

### PR DESCRIPTION
**Pull request by bduffany**
In some places where we check that the user has access to a particular group, we'll now want to check whether they're in the group _and_ they fulfill a certain role within the group. This makes the change to also select the `role` field from `UserGroup` in all the places where we already join with the `UserGroups` table for group memberships.

---

**Version bump**: Patch

### CODE REVIEW

File: enterprise/server/auth/auth.go [hunk 1]
```diff
@@ -472,19 +481,46 @@ func lookupUserFromSubID(env environment.Env, ctx context.Context, subID string)
...
+			  g.user_id,
+				g.group_id,
+				g.url_identifier,
+				g.name,
+				g.owned_domain,
+				g.github_token,
+				g.sharing_enabled,
+				g.use_group_owned_executors,
+				g.saml_idp_metadata_url,
+				ug.role
```
Comment (bduffany) [inline enterprise/server/auth/auth.go:495]: FYI this is selecting all the fields from Group except

* Model fields (created_at_usec, updated_at_usec) -- we don't reference these anywhere
* WriteToken and APIKey -- these aren't used anywhere either; they can be removed in a separate PR

File: enterprise/server/auth/auth.go [hunk 2]
```diff
@@ -472,19 +479,46 @@ func lookupUserFromSubID(env environment.Env, ctx context.Context, subID string)
 		if err := userRow.Take(user).Error; err != nil {
 			return err
 		}
-		groupRows, err := tx.Raw(`SELECT g.* FROM `+"`Groups`"+` as g JOIN UserGroups as ug
-                                          ON g.group_id = ug.group_group_id
-                                          WHERE ug.user_user_id = ?`, user.UserID).Rows()
+		groupRows, err := tx.Raw(`
+			SELECT
+			  g.user_id,
```
Comment (vadimberezniker) [inline enterprise/server/auth/auth.go:484]: nit: fix indent

File: enterprise/server/auth/auth.go [hunk 3]
```diff
@@ -117,6 +120,10 @@ func (c *Claims) GetAllowedGroups() []string {
 	return c.AllowedGroups
 }
 
+func (c *Claims) GetGroupRoles() []uint32 {
```
Comment (vadimberezniker) [inline enterprise/server/auth/auth.go:123]: Can we combine the group ID & role into a single struct instead of returning them separately? I don't think the user of this API should have to lookup this data in two slices. 

Same for the Claims struct, is there a reason not to store the per-group data (ID + role) together? We could introduce a new field and remove the old allowedgroups field later once it's no longer used.

File: enterprise/server/auth/auth.go (hunk 2 above)
Comment (bduffany) [reply enterprise/server/auth/auth.go:484]: This indent is intentional since the field list is syntactically a "child" of the SELECT

File: enterprise/server/auth/auth.go (hunk 3 above)
Comment (bduffany) [reply enterprise/server/auth/auth.go]: Done

File: enterprise/server/auth/auth.go [hunk 4]
```diff
@@ -98,11 +100,12 @@ func jwtKeyFunc(token *jwt.Token) (interface{}, error) {
...
 type Claims struct {
 	jwt.StandardClaims
-	UserID                 string                   `json:"user_id"`
-	GroupID                string                   `json:"group_id"`
-	AllowedGroups          []string                 `json:"allowed_groups"`
-	Capabilities           []akpb.ApiKey_Capability `json:"capabilities"`
-	UseGroupOwnedExecutors bool                     `json:"use_group_owned_executors,omitempty"`
+	UserID                 string                        `json:"user_id"`
+	GroupID                string                        `json:"group_id"`
+	AllowedGroups          []string                      `json:"allowed_groups"`
```
Comment (vadimberezniker) [inline enterprise/server/auth/auth.go:105]: Add a TODO to remove this field.

File: enterprise/server/auth/auth.go (hunk 2 above)
Comment (vadimberezniker) [reply enterprise/server/auth/auth.go]: The indentation you had is good. I was pointing out that for some reason the indentation for the first field was different from the remaining fields.

**Review by vadimberezniker — APPROVED** [review]

File: enterprise/server/auth/auth.go (hunk 2 above)
Comment (bduffany) [reply enterprise/server/auth/auth.go]: Gotcha, I think it was indented with spaces instead of tabs so I didn't notice it locally. Done

File: enterprise/server/auth/auth.go (hunk 4 above)
Comment (bduffany) [reply enterprise/server/auth/auth.go:106]: Done
